### PR TITLE
feat: add `AtomicOp` cases to `Verifier` (records, errors, assign)

### PR DESCRIPTION
**Pull request by felix-berg**
This PR has the errors in the `CompilerSuite` regarding the cases `RecordRestrict` and `RecordSelect`.

### CODE REVIEW

**Comment by JonathanStarup** [comment]
There's a bug with record pattern matching in simplifier, i'll give it a go once I get to it on my todo list

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 1]
```diff
@@ -302,6 +302,49 @@ object Verifier {
...
+          val List(t1, t2) = ts
+          t1 match {
+            case MonoType.Ref(elm) =>
+              checkEq(t2, elm, loc)
+              check(expected = MonoType.Unit)(actual = tpe, loc)
+            case _ => failMismatchedShape(t1, "Ref", loc)
+          }
+
+        // Match- and Hole-errors match with any type
+        case AtomicOp.HoleError(_) | AtomicOp.MatchError => tpe
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:315]: For pattern matches we usually format everything consistently, so a single case per case so to speak, and newline after `=>` and before `case`

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 2]
```diff
@@ -302,6 +302,49 @@ object Verifier {
...
+              checkEq(valtype, t1, loc)
+              tpe
+            case (_, None) => failMismatchedShape(tpe, s"Record with ${label.name}", loc)
+          }
+
+        case AtomicOp.RecordRestrict(label) =>
+          val List(t1) = ts
+          removeFromRecordType(t1, label.name, loc) match {
+            case (rec, Some(_)) =>
+              // TODO: VERIFIER: this fails when a is removed from { a = Bool | { a = Int32 | { b = Int32 | {}}}}
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:332]: These errors should go away after #7479 :)

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 3]
```diff
@@ -302,6 +302,49 @@ object Verifier {
...
+          }
+
+        // Match- and Hole-errors match with any type
+        case AtomicOp.HoleError(_) =>
+          tpe
+
+        case AtomicOp.MatchError =>
+          tpe
+
+        case AtomicOp.RecordEmpty => check(expected = MonoType.RecordEmpty)(actual = tpe, loc)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:321]: oh,  missing newline after => and before next case

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 4]
```diff
@@ -416,6 +459,27 @@ object Verifier {
     else failMismatchedTypes(tpe1, tpe2, loc)
   }
 
+  private def removeFromRecordType(rec: MonoType, label: String, loc: SourceLocation): (MonoType, Option[MonoType]) = rec match {
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:462]: add doc comments to the helper functions

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**